### ISSUE TITLE
Add station metadata download

### ISSUE THREAD

**Issue by rod-glover** [pacificclimate/station-data-portal-backend#5]
Option: Use GeoServer as backend for this, like existing PDP PCDS apps does. This has the advantage of providing many formats out of the box, and likely low effort to set up.

TODO:

- [ ] Examine server logs to determine how often and who downloads the station metadata in bulk:
    * frequency
    * user identities
    * formats
- [ ] Decide if worth implementing. If so:
    * what formats?
    * method?
        * if few formats, ... ?
        * if many formats, GeoServer, probably

**Comment by rod-glover** [comment]
# Server log research

Backend calls for bulk metadata downloads are distinguished by either of these conditions

1. WMS
    `url ~ /geoserver/CRMP/ows`
    `service === WMS`
    `request === GetMap`

2. WFS
    `url ~ /geoserver/CRMP/ows`
    `service === WFS`
    `outputFormat !== json` 
     * JSON output format is used by the View Metadata button, so this could be a mix of download and view.


## WMS

```bash
root@pizza:/home/websites/tools.pacificclimate.org/logs/old# cat metadata-wms.sh 
#!/bin/bash
for fn in ./ssl-access.log.*.gz; do
    zgrep --ignore-case "/geoserver/CRMP/ows\?.*service=WMS.*request=GetMap" ${fn}
done

root@pizza:/home/websites/tools.pacificclimate.org/logs/old# ./metadata-wms.sh >metadata-wms.out

root@pizza:/home/websites/tools.pacificclimate.org/logs/old# wc -l metadata-wms.out 
0 metadata-wms.out
```
**Conclusion**: No one in recent history has requested any metadata in a WMS format.

## WFS

```bash
root@pizza:/home/websites/tools.pacificclimate.org/logs/old# cat metadata-wfs.sh
#!/bin/bash
for fn in ./ssl-access.log.*.gz; do
    zgrep --ignore-case "/geoserver/CRMP/ows\?.*service=WFS" ${fn}
done

root@pizza:/home/websites/tools.pacificclimate.org/logs/old# ./metadata-wfs.sh >metadata-wfs.out

root@pizza:/home/websites/tools.pacificclimate.org/logs/old# wc -l metadata-wfs.out 
590 metadata-wfs.out

root@pizza:/home/websites/tools.pacificclimate.org/logs/old# cut -d "&" -f 5 metadata-wfs.out | sort | uniq
outputFormat=csv
outputFormat=json
outputFormat=SHAPE-ZIP

root@pizza:/home/websites/tools.pacificclimate.org/logs/old# grep "outputFormat=csv" metadata-wfs.out | wc -l
85

root@pizza:/home/websites/tools.pacificclimate.org/logs/old# grep "outputFormat=json" metadata-wfs.out | wc -l
482

root@pizza:/home/websites/tools.pacificclimate.org/logs/old# grep "outputFormat=SHAPE-ZIP" metadata-wfs.out | wc -l
23
```

**Conclusion**: Assuming JSON downloads are all View Metadata button clicks, there are relatively few (~20%) bulk metadata downloads, and those are all either CSV or SHAPE-ZIP.

**Comment by rod-glover** [comment]
Conclusions: 
* Only CSV, SHAPE-ZIP, and (possibly) JSON formats are in demand.
* On average, a metadata download occurs a bit less than 2x / day.
* 57 unique IP addresses request non-JSON downloads.

**Comment by rod-glover** [comment]
I got curious about all those `image/png` requests. I generated a file (`metadata-wms.png.out`) containing only those requests.

```bash
1417174 metadata-wms.png.out
root@pizza:/home/websites/tools.pacificclimate.org/logs/old# awk -F' |&|\?' '{print $1}' metadata-wms.png.out | sort | uniq
```

There were >1.4 million such requests in the last year.
* How is this even possible?
* It seems unlikely that there were 1.4 M bulk downloads of metadata.
* So what is happening here?

```bash
root@pizza:/home/websites/tools.pacificclimate.org/logs/old# awk -F' |&|\?' '{print $1}' metadata-wms.png.out | sort | uniq | wc -l
1893
```
These requests are coming from _far_ more (1893) IP's than other kinds.

**Comment by rod-glover** [comment]
`image/png` requests are for the map proper and are not surprising in number or origin.

**Comment by rod-glover** [comment]
@jameshiebert , it looks to me as if there is enough demand to justify using Geoserver to generate the metadata. I was hoping to get away with just JSON, but I think that would leave some people out in the cold. What do you think?

**Comment by jameshiebert** [comment]
Yeah, I mean that's pretty low usage for the feature, however, I don't think that it's reasonable to even publish the station data *without* providing this feature in some form. So I think we're committed to it.

And yes, there's enough variety there that it probably makes sense to let a Geoserver deployment handle this. It *should* be pretty simple... just deploy a container with Geoserver and a config that points to `crmp_network_geoserver` view in the database.
